src/boot/store.js

```diff
@@ -118,12 +119,57 @@ const migrations: { [string]: (GlobalState) => GlobalState } = {
       ackedPushToken: null,
     })),
   }),
+
+  '10': state => ({
+    ...state,
+    accounts: state.accounts.map(a => ({
+      ...a,
+      zulipVersion:
+        // $FlowMigrationFudge
+        a.zulipVersion instanceof ZulipVersion ? a.zulipVersion : new ZulipVersion(a.zulipVersion),
+    })),
+  }),
 };
 
+/**
+ * Persist ZulipVersion instances as raw version string.
+ */
+const zulipVersionTransform = createTransform(
```

Comment:
> this is a breeding ground for bugs.

It would be, as-is, but this PR is WIP to decide if we want to take this approach or the radically different one I mentioned [here](https://github.com/zulip/zulip-mobile/issues/3951#issuecomment-594790134). I hope this code is informative enough to help make that decision, even if the lack of type-checking means it has bugs as-is.